tests/test_octodns_provider_ultra.py

```diff
@@ -55,7 +64,8 @@ def test_login(self):
             self.assertEquals(1, mock.call_count)
             expected_payload = "grant_type=password&username=user&"\
                                "password=rightpass"
-            self.assertEquals(mock.last_request.text, expected_payload)
+            self.assertEquals(parse_qs(mock.last_request.text),
+                              parse_qs(expected_payload))
```

Comment:
What error lead to needing to parse the query strings and include `urlparse` up above?

Reply:
This was the failing test:

```
======================================================================
FAIL: test_login (test_octodns_provider_ultra.TestUltraProvider)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/blanariu/work/cst/octodns/tests/test_octodns_provider_ultra.py", line 60, in test_login
    self.assertEquals(mock.last_request.text, expected_payload)
AssertionError: u'username=user&password=rightpass&grant_type=password' != u'grant_type=password&username=user&password=rightpass'
- username=user&password=rightpass&grant_type=password
+ grant_type=password&username=user&password=rightpass
```

I believe it has to do with the insert order preservation differences between 2.7 and 3.x

Reply:
Ah. Yeah. 3.x preserves insertion order, 2.x it's undefined/random. Odd that I can't get it to fail. This bit makes sense. Please go ahead and apply the `six` suggestion above.